nav2_dwb_controller/dwb_plugins/src/kinematic_parameters.cpp

```diff
@@ -58,7 +58,7 @@ bool KinematicParameters::isValidSpeed(double x, double y, double theta)
   if (max_speed_xy_ >= 0.0 && vmag_sq > max_speed_xy_sq_ + EPSILON) {return false;}
   if (min_speed_xy_ >= 0.0 && vmag_sq + EPSILON < min_speed_xy_sq_ &&
     min_speed_theta_ >= 0.0 && fabs(theta) + EPSILON < min_speed_theta_) {return false;}
```

Comment:
I think the easier solution is to move all of the `isValidSpeed` method out of the kinematic parameters and into the users of it. Please move `isValidSpeed` into the XY iterator and call that version of it. I think that would fix this issue and be a little cleaner.

Reply:
`isValidSpeed` method is moved to the XY iterator